### PR TITLE
Allow disable skipping study transformation with meta_study.txt file

### PR DESCRIPTION
**Pull request by oplantalech**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -260,6 +260,7 @@ files are not removed from the `scan.location`.
 - `transformation.command.script.docker.image`: if the transformation command runs in Docker, add here the Docker image name. The app will call the script in the path provided in `transformtion.command.script`. If your command is executed at the entrypoint, you can comment `transformation.command.script` out.
 - `transformation.skip`: set this parameter to `true` if you want to skip the transformation step.
 - `transformation.directory`: resource path to directory where transformed study files are placed after transformation. When not set, transformed files are placed in the 'staging' subdirectory of study folders in the `etl.working.dir`.
+- `transformation.no.meta.check`: set this parameter to to `true` if you want to transform also studies containing a meta_study.txt file (skipped by default even if `transformation.skip` is `false`).
```
Comment (pvannierop) [inline README.md:263]: I get lost in the double negatives in `no` and `false` as default. How about making the option `transformation.metafile.check` that defaults to `true`?

File: src/main/java/org/cbioportal/staging/etl/Transformer.java [hunk 2]
```diff
@@ -32,13 +32,17 @@
...
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.core.io.Resource;
 import org.springframework.stereotype.Component;
 
 @Component
 public class Transformer {
     private static final Logger logger = LoggerFactory.getLogger(Transformer.class);
 
+    @Value("${transformation.no.meta.check:false}")
```
Comment (pvannierop) [inline src/main/java/org/cbioportal/staging/etl/Transformer.java:43]: Make `transformation.metafile.check` with default `true`.

File: src/main/java/org/cbioportal/staging/etl/Transformer.java [hunk 3]
```diff
@@ -74,7 +78,7 @@
                     Resource logFile = utils.createFileResource(transformedFilesPath, study.getStudyId() + "_transformation_log.txt");
                     logFiles.put(study, logFile);
 
-                    if (metaFileExists(untransformedFilesPath)) {
+                    if (transformationNoMetaCheck && metaFileExists(untransformedFilesPath)) {
```
Comment (pvannierop) [inline src/main/java/org/cbioportal/staging/etl/Transformer.java:81]: This condition is not correct I think. When making the prop `transformation.metafile.check`:

```
if (metaFileCheck && metaFileExists(untransformedFilesPath)) {
... --> then skip

```

File: src/main/resources/application.properties [hunk 4]
```diff
@@ -84,6 +84,8 @@ transformation.skip=true
 # (optional) Directory to save the transformed files. If it is not set, the transformed files will be saved in the working directory,
 # in a folder "staging" inside each study-specific folder.
 #transformation.directory=
+# (optional) Set this to true if you want to transform also studies containing a meta_study.txt file
+#transformation.no.meta.check=true
```
Comment (pvannierop) [inline src/main/resources/application.properties:88]: Update to remarks above

File: README.md (hunk 1 above)
Comment (oplantalech) [reply README.md:263]: @pvannierop no problem. I also had doubts when implementing it this way.

**Pull request merged**